### PR TITLE
fix: add funding agreement buttons yes no

### PR DESCRIPTION
**Pull request by gurjmatharu**

### CODE REVIEW

File: app/components/Form/ProjectFundingAgreementForm.tsx [hunk 1]
```diff
@@ -119,13 +121,28 @@ const ProjectFundingAgreementForm: React.FC<Props> = (props) => {
     handleChange(fundingAgreement.newFormData, "staged");
   };
 
+  const router = useRouter();
+  const navigateToReviewAndSubmit = () => {
+    console.log("navigateToReviewAndSubmit");
```
Comment (pbastia) [inline app/components/Form/ProjectFundingAgreementForm.tsx:126]: this can be deleted

File: app/components/Form/ProjectFundingAgreementForm.tsx [hunk 2]
```diff
@@ -119,13 +121,28 @@ const ProjectFundingAgreementForm: React.FC<Props> = (props) => {
...
+          <h3>Is this a funded project?</h3>
+          <Button onClick={addFundingAgreement} style={{ marginRight: "1rem" }}>
+            Yes
+          </Button>
+          <Button
+            onClick={navigateToReviewAndSubmit}
+            style={{ marginRight: "1rem" }}
+          >
+            No
+          </Button>
```
Comment (pbastia) [inline app/components/Form/ProjectFundingAgreementForm.tsx:144]: These should be radio buttons, conditionally rendering the form whether or not 'yes' is selected or not
@nanyangpro will provide a very quick wireframe today

File: schema/data/dev/004_cif_project.sql [hunk 3]
```diff
@@ -38,14 +38,23 @@ do $$
...
+        json_build_object(
           'contactId', index,
-          'projectId', index,
+          'projectId', (select form_data_record_id from cif.form_change where form_data_table_name='project' and project_revision_id=current_revision.id),
           'contactIndex', 1
-      )
-      where project_revision_id=current_revision.id and form_data_table_name='project_contact';
-
+          ),
+        'create', 'cif', 'project_contact', 'pending', 'project_contact', current_revision.id);
```
Comment (pbastia) [inline schema/data/dev/004_cif_project.sql:57]: this is not part of this PR

**Review by pbastia — CHANGES_REQUESTED** [review]

File: schema/data/dev/004_cif_project.sql (hunk 3 above)
Comment (gurjmatharu) [reply schema/data/dev/004_cif_project.sql:57]: i've rebased on the other PR should be fixed once the other one is merged to dev.

**Review by pbastia — CHANGES_REQUESTED** [review]
This PR is missing the part where we show the radio buttons when the form is displayed, as in [the wireframe](https://www.figma.com/proto/PzOY8RyQnVXnjZJdSv2AqQ/Wireframing_Figma?page-id=2655%3A2669&node-id=3426%3A3149&viewport=-1133%2C1340%2C0.2&scaling=contain&starting-point-node-id=3426%3A2678&show-proto-sidebar=1)
(we can leave the "Undo Changes" button + saving indicator just below the form title, for consistency with all the other forms)

File: app/components/Form/ProjectFundingAgreementForm.tsx [hunk 4]
```diff
@@ -118,19 +123,62 @@ const ProjectFundingAgreementForm: React.FC<Props> = (props) => {
...
-        <Button onClick={addFundingAgreement} style={{ marginRight: "1rem" }}>
-          Add Funding Agreement
-        </Button>
+        <div>
+          <h3>Is this a funded project?</h3>
+          <RadioButton
+            name="yes"
+            label="Yes"
+            className="radio-button"
+            onClick={addFundingAgreement}
```
Comment (pbastia) [inline app/components/Form/ProjectFundingAgreementForm.tsx:150]: This is a double-click bug, we should set a `disabled` field that depends on the `isAddingFundingParameterFormChange` property

File: app/components/Form/ProjectFundingAgreementForm.tsx [hunk 5]
```diff
@@ -118,19 +123,62 @@ const ProjectFundingAgreementForm: React.FC<Props> = (props) => {
...
+              onClick={() => setShowDiscardConfirmation(true)}
+            />
+            {showDiscardConfirmation && (
+              <Alert variant="danger" size="sm">
+                All changes made will be deleted.
+                <a onClick={handleDiscard} id="confirm-discard-revision">
+                  Proceed
+                </a>
+                <a onClick={() => setShowDiscardConfirmation(false)}>Cancel</a>
+              </Alert>
```
Comment (pbastia) [inline app/components/Form/ProjectFundingAgreementForm.tsx:178]: This should be refactored in its own component, we have the exact same code and styles in `[projectRevision]/index.tsx`

File: schema/deploy/mutations/discard_funding_parameter_form_change.sql [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+begin;
+
+/**
+  Removing or archiving a funding_parameter is a chained operation. The data for funding_parameter is spread across two tables:
+    - reporting_requirement (base table, common to all reports)
+    - funding_parameter (data specific to parameter_form)
+  Because this data is spread across two tables we have to remove or archive two form_change records within one transaction, one for each table.
+**/
+
+create or replace function cif.discard_funding_parameter_form_change(revision_id int)
```
Comment (pbastia) [inline schema/deploy/mutations/discard_funding_parameter_form_change.sql:14]: We need tests for this function

**Review by pbastia — CHANGES_REQUESTED** [review]
Looks great! Just a couple of things to refactor to keep the code cleaner

File: app/components/Form/ProjectFundingAgreementForm.tsx [hunk 7]
```diff
@@ -118,19 +124,61 @@ const ProjectFundingAgreementForm: React.FC<Props> = (props) => {
...
+          />
+          <RadioButton name="yes" label="No" className="radio-button" />
+        </div>
       )}
       {projectRevision.projectFundingAgreementFormChanges.edges.length > 0 && (
         <>
+          <div>
+            <h3>Is this a funded project?</h3>
+            <RadioButton
+              name="yes"
```
Comment (pbastia) [inline app/components/Form/ProjectFundingAgreementForm.tsx:162]: Can we get the radio button group a better name? 😁

File: app/components/Form/ProjectFundingAgreementForm.tsx [hunk 8]
```diff
@@ -118,19 +124,61 @@ const ProjectFundingAgreementForm: React.FC<Props> = (props) => {
...
+              label="Yes"
+              className="radio-button"
+              checked={true}
+            />
+            <RadioButton
+              name="yes"
+              label="No"
+              className="radio-button"
+              onClick={() => setShowDiscardConfirmation(true)}
+              disabled={isAddingFundingParameterFormChange}
```
Comment (pbastia) [inline app/components/Form/ProjectFundingAgreementForm.tsx:172]: This should be `isDiscardingFundingParameterFormChange` that can be retrieved from the mutation hook

File: app/lib/theme/DangerAlert.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { Alert } from "@button-inc/bcgov-theme";
+
+const DangerAlert: React.FC<{ proceedOnClick; cancelOnClick }> = ({
```
Comment (pbastia) [inline app/lib/theme/DangerAlert.tsx:4]: we should use a real interface and types for these methods. We can also give them better names:

```tsx
interface Props {
  onProceed: () => void;
  onCancel: () => void;
}

const DangerAlert: React.FC<Props> = ...
```

File: app/tests/unit/pages/project-revision/[projectRevision]/index.test.tsx [hunk 10]
```diff
@@ -321,7 +321,7 @@ describe("The Create Project page", () => {
     act(() => userEvent.click(screen.queryByText("Discard Project Revision")));
 
     expect(
-      screen.getByText(/All changes made will be permanently deleted./)
+      screen.getByText(/All changes made will be deleted./)
```
Comment (pbastia) [inline app/tests/unit/pages/project-revision/[projectRevision]/index.test.tsx:324]: The wording here was approved by our design team, we shouldn't be changing it. Maybe make it configurable in the `DangerAlert` component?

File: app/lib/theme/DangerAlert.tsx [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+            margin-left: 2em;
+            margin-right: 1em;
+          }
+        `}
+      </style>
+    </div>
+  );
+};
+
+export default DangerAlert;
```
Comment (pbastia) [inline app/lib/theme/DangerAlert.tsx:52]: We also might want a better name for this component, like `ConfirmationAlert`

**Review by pbastia — CHANGES_REQUESTED** [review]
Very nice! Just a few naming and style conventions, looks great!

File: app/components/Form/ProjectFundingAgreementForm.tsx [hunk 12]
```diff
@@ -118,19 +126,67 @@ const ProjectFundingAgreementForm: React.FC<Props> = (props) => {
...
+            label="Yes"
+            className="radio-button"
+            onClick={addFundingAgreement}
+            disabled={isDiscardingFundingParameterFormChange}
+          />
+          <RadioButton
+            name="skip-funding-agreement"
+            label="No"
+            className="radio-button"
+          />
```
Comment (pbastia) [inline app/components/Form/ProjectFundingAgreementForm.tsx:160]: radio buttons need to share the same name when in the same group

**Review by pbastia — CHANGES_REQUESTED** [review]

**Review by pbastia — APPROVED** [review]

**Review by pbastia — APPROVED** [review]
:shipit:
